src/cbor/decoder.rs

```diff
@@ -139,7 +139,7 @@ impl DecoderCursor {
                 CborType::Tag(tag, Box::new(item))
             }
             7 => self.read_null()?,
-            _ => unreachable!(),
+            _ => return Err(CborError::MalformedInput),
```

Comment:
I must be misunderstanding something about rust - how is this reachable? (I guess, to put it another way, we should also include the test input that triggered this)

Reply:
I added a test case. We're missing major type 3 at the moment (strings). Otherwise the compiler would complain about unreachable code in `_` 🙂